editor/block-settings-menu/unknown-converter.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton } from '@wordpress/components';
+import { getUnknownTypeHandlerName, rawHandler, serialize } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { getBlock } from '../selectors';
+import { replaceBlocks } from '../actions';
+
+export function UnknownConverter( { block, convertToBlocks, small } ) {
+	if ( ! block || getUnknownTypeHandlerName() !== block.name ) {
+		return null;
+	}
+
+	const label = __( 'Convert to blocks' );
+
+	return (
+		<IconButton
+			className="editor-block-settings-menu__control"
+			onClick={ () => convertToBlocks( block ) }
+			icon="screenoptions"
+			label={ small ? label : undefined }
+		>
+			{ ! small && label }
+		</IconButton>
+	);
+}
+
+export default connect(
+	( state, { uid } ) => ( {
+		block: getBlock( state, uid ),
+	} ),
+	( dispatch, { uid } ) => ( {
```

Comment:
You have `block` already exposed in `props`. The following should work:

```js
( dispatch, { block, uid } ) => ( {
convertToBlocks() {
	dispatch( replaceBlocks( uid, rawHandler( {
		HTML: serialize( block ),
		inline: false,
	} ) ) );
},
```

Reply:
Another funky way to do the same is to wrap this with HOC `withBlock` :P

```js
const withBlock = component => connect(  ( state, { uid } ) => ( {
	block: getBlock( state, uid ),
} ) )( component );

export default connect(
	null,
	( dispatch, { block, uid } ) => ( {
		convertToBlocks() {
			dispatch( replaceBlocks( uid, rawHandler( {
				HTML: serialize( block ),
				inline: false,
			} ) ) );
		},
	} )
)( withBlock( UnknownConverter ) );

Reply:
I'm fine with keeping as is :)

Reply:
👍